Bottom: c85dc6134b42d7468ba5f8deff7e303a14c39a76
Top:    e4e59e8f8f15c02a0607f59cf38a6c1f08d6d043
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-04-20 11:54:18 -0400

Refresh of tas5504-codec

---

diff --git a/sound/soc/codecs/tas5504.c b/sound/soc/codecs/tas5504.c
index 6ac3289..d2dbdaa 100644
--- a/sound/soc/codecs/tas5504.c
+++ b/sound/soc/codecs/tas5504.c
@@ -820,9 +820,11 @@ static int tas5504_i2c_probe(struct i2c_client *client,
 	INIT_LIST_HEAD(&priv->codec.dapm_widgets);
 	INIT_LIST_HEAD(&priv->codec.dapm_paths);
 
+#if defined(CONFIG_SND_SOC_OF_SIMPLE)
 	/* Tell the of_soc helper about this codec */
-	of_snd_soc_register_codec(&tas5504_soc_codec_dev, priv, &tas5504_dai,
+	of_snd_soc_register_codec(&tas5504_soc_codec_dev, priv, &tas5504_dai, 1,
 				  client->dev.archdata.of_node);
+#endif
 
 	dev_dbg(&client->dev, "I2C device initialized\n");
 	return 0;
